Assets/Scripts/Managers/PlacementSystem/PlacementManager.cs

```diff
@@ -150,8 +152,22 @@ private void OnPlaceStructure() {
 		if (_canPlace) {
 			GameObject newPlacement = Instantiate(_buildablePrefab, _currentPlacedPrefab.transform.position, _currentPlacedPrefab.transform.rotation);
 			newPlacement.transform.parent = _placementPosition.transform;
+			newPlacement.AddComponent<MeshCollider>();
+			newPlacement.layer = 9;
 
 			EventBus.Instance.TriggerEvent<GameObject>(EventType.CHANGE_STRUCTURE, _buildablePrefab);
 		}
 	}
+
+	private void OnRemoveStructure() {
+		int layer = 9;
+		int layerMask = 1 << layer;
+		Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
```

Comment:
Maybe absctract this behaviour into a seperate function later?
We now have both the OnMoveStructure and OnRemoveStructure which use the same behaviour aka duplicate code.
Doesn't need attention right now but might be a good improvement in the future.